### PR TITLE
KYLIN-4413: add canary tool

### PR DESCRIPTION
**Pull request by zhoukangcn**
https://issues.apache.org/jira/browse/KYLIN-4413

### CODE REVIEW

File: tool/src/main/java/org/apache/kylin/tool/KylinCanary.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+        return response;
+    }
+
+    private long checkQuery(String host) throws IOException {
+        long startTime = System.currentTimeMillis();
+        String url = "http://" + host + "/kylin/api/query";
+
+        HttpResponse response = makeHttpRequest(url,
+                "{\"sql\": \"SELECT 1\",\"project\": \"KYLIN_SYSTEM\"}");
```
Comment (shaofengshi) [inline tool/src/main/java/org/apache/kylin/tool/KylinCanary.java]: The project "KYLIN_SYSTEM" may not exist ?

File: tool/src/main/java/org/apache/kylin/tool/KylinCanary.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            logger.warn("new class: " + sinkClassName + " error.", e);
+            sink = new StdOutSink();
+        }
+        sink.init(null);
+
+        metricsSystem = DefaultMetricsSystem.instance();
+
+        metricsSystem.init("kylin-canary");
+        metricsSystem.register("kylin-canary", "Kylin Canary", canaryMetrics);
+        metricsSystem.register("falcon", "Falcon sink for xiaomi", sink);
```
Comment (shaofengshi) [inline tool/src/main/java/org/apache/kylin/tool/KylinCanary.java]: Change a general name?

File: core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java [hunk 3]
```diff
@@ -2475,4 +2475,21 @@ public int getStaleJobThresholdInDays() {
...
+    public boolean isCanaryDaemon() {
+        return Boolean.parseBoolean(getOptional("kylin.canary.daemon", FALSE));
+    }
+
+    public long getCanaryInterval() {
+        return Long.parseLong(getOptional("kylin.canary.interval", "6000"));
+    }
+
+    public String getCanaryToken() {
+        return getOptional("kylin.canary.token", "");
```
Comment (shaofengshi) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java]: Keep the basic auth in configuration file will increase the credential exposure risk. Is there a better way to do the authentication?

File: tool/src/main/java/org/apache/kylin/tool/KylinCanary.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    private long checkQuery(String host) throws IOException {
+        long startTime = System.currentTimeMillis();
+        String url = "http://" + host + "/kylin/api/query";
+
+        HttpResponse response = makeHttpRequest(url,
+                "{\"sql\": \"SELECT 1\",\"project\": \"KYLIN_SYSTEM\"}");
+
+        int code = response.getStatusLine().getStatusCode();
```
Comment (shaofengshi) [inline tool/src/main/java/org/apache/kylin/tool/KylinCanary.java]: Using org.apache.kylin.common.restclient.RestClient might be a better solution

**Comment by zhoukangcn** [comment]
> How to use this canary tool? Seems it need to be integrated with some monitoring tool. And, is this an extensible design so that more test can be added to it later?

Thank you @shaofengshi 
I will add some tests later.

**Pull request closed without merging**